sql/types/enum.go

```diff
@@ -278,15 +280,17 @@ func (t EnumType) CollationCoercibility(ctx *sql.Context) (collation sql.Collati
 
 // Zero implements Type interface.
 func (t EnumType) Zero() interface{} {
-	// / If an ENUM column is declared NOT NULL, its default value is the first element of the list of permitted values.
-	return uint16(1)
+	// TODO: If an ENUM column is declared NOT NULL, its default value is the first element of the list of permitted values.
+	return uint16(0)
 }
 
 // At implements EnumType interface.
 func (t EnumType) At(index int) (string, bool) {
-	// / The elements listed in the column specification are assigned index numbers, beginning with 1.
+	// The elements listed in the column specification are assigned index numbers, beginning with 1.
 	index -= 1
-	if index < 0 || index >= len(t.indexToVal) {
+	if index == -1 {
```

Comment:
(minor) Might want to leave this one `index <= -1`, just in case some code ever incorrectly passed in a negative integer value to this function. Probably not likely to happen, but might as well catch that case while we're in here.